### PR TITLE
Fix issues reported by static analyzers.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Http/CallbackStream.php [hunk 1]
```diff
@@ -40,7 +40,8 @@ public function getContents(): string
     {
         $callback = $this->detach();
         $result = '';
-        if (is_callable($callback)) {
+        /** @psalm-suppress TypeDoesNotContainType */
+        if ($callback !== null) {
```
Comment (othercorey) [inline src/Http/CallbackStream.php:44]: Why is this needed?

Comment (ADmad) [reply src/Http/CallbackStream.php:44]: The return type of `detach()` is `?callable` so checking `!== null` is cheaper hence I made the change. Pslam gave some silly error like `Type null for $callback is always null` which I ignored.

File: src/I18n/DateFormatTrait.php [hunk 2]
```diff
@@ -251,7 +251,7 @@ protected function _formatObject($date, $format, ?string $locale): string
                 $calendar,
                 $pattern
             );
-            if ($formatter === false) {
+            if (empty($formatter)) {
```
Comment (ADmad) [inline src/I18n/DateFormatTrait.php:254]: psalm says return value of `datefmt_create()` can be false while phpstan says it can be `null`. It seems the PHP doc was updated at some point (or changed for PHP 8). Regardless, using `empty()` caters for both types.

File: src/Http/CallbackStream.php (hunk 1 above)
Comment (othercorey) [reply src/Http/CallbackStream.php:44]: It probably doesn't consider callable a native type and discards it. Probably a bug.
